### PR TITLE
[0.0.2-beta.113] Add qa preview actions proto

### PR DESCRIPTION
**Pull request by zrg-team**
## Description

+ Add QA preview proto

## Issues

https://jitera.atlassian.net/browse/JAIS-368

# Checklist

- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation to reflect the change
- [ ] I have updated the README.md to reflect the change
- [ ] I have added unit tests to prove that the update is effective and the feature works

### CODE REVIEW

File: protos/v2/project_source.proto [hunk 1]
```diff
@@ -55,6 +55,7 @@ message Git {
 	string branch = 5;
 	string token = 6;
 	string jitera_branch = 7;
+	string target_branch = 8;
```
Comment (comictvn) [inline protos/v2/project_source.proto:58]: what's target branch? @zrg-team . What's information of git you need?

File: protos/v2/project_preview.proto [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  int32 token_usage = 5;
+
+  enum Status {
+    SUCCEEDED = 0;
+    INPROGRESS = 1;
+    FAILED = 2;
+  }
+}
+
+message DeployGenerateScriptItem {
```
Comment (comictvn) [inline protos/v2/project_preview.proto:33]: Do we need this at the moment?

File: protos/v2/project_source.proto (hunk 1 above)
Comment (comictvn) [reply protos/v2/project_source.proto:58]: I see target branch is pull request branch

File: protos/v2/project_preview.proto (hunk 2 above)
Comment (zrg-team) [reply protos/v2/project_preview.proto:33]: yes i think. We need AI to generate script to deploy right ?

File: protos/v2/project_source.proto (hunk 1 above)
Comment (zrg-team) [reply protos/v2/project_source.proto:58]: I mean 
`branch`: user select branch
`jitera_branch`: base branch that use to push ERD
`target_branch `: branch that users want to do something with. Example feat/uc-1

File: protos/v2/project_preview.proto (hunk 2 above)
Comment (comictvn) [reply protos/v2/project_preview.proto:33]: oh deployment script. yah I missed it

**Review by comictvn — APPROVED** [review]

**Review by JoelNguyen1608 — APPROVED** [review]

**Review by jitera-carrick — APPROVED** [review]

File: protos/v2/project_source.proto (hunk 1 above)
Comment (jitera-carrick) [reply protos/v2/project_source.proto:58]: @zrg-team We need an attribute: `uc_pr_name`. Ultron need it to create PR. @JoelNguyen1608

**Review by comictvn — CHANGES_REQUESTED** [review]
@zrg-team you might forgot require new project_preview file in this file ruby/lib/jitera_proto.rb

File: protos/v2/project_preview.proto [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+message ProjectPreviewDeployGenerateRequest {
+  int32 project_preview_id = 1;
+  int32 project_id = 2;
+  ProjectSource project_source = 3;
+  Git git = 4;
+  repeated DeployGenerateScriptItem previous_scripts = 5;
+}
+
+message ProjectPreviewDeployGenerateResponse {
```
Comment (comictvn) [inline protos/v2/project_preview.proto:46]: we need docker base image like ruby:3.0.3-alpine3.15, node:18-alpine based on user import project.

File: protos/v2/project_preview.proto [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  string command = 1;
+  string description = 2;
+}
+
+message ProjectPreviewDeployGenerateRequest {
+  int32 project_preview_id = 1;
+  int32 project_id = 2;
+  ProjectSource project_source = 3;
+  Git git = 4;
+  repeated DeployGenerateScriptItem previous_scripts = 5;
```
Comment (comictvn) [inline protos/v2/project_preview.proto:43]: it will be blank at the first time?

Comment (zrg-team) [reply protos/v2/project_preview.proto:43]: yes

File: ruby/lib/payloads/v2/project_preview_pb.rb [hunk 5]
```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+# Generated by the protocol buffer compiler.  DO NOT EDIT!
+# source: v2/project_preview.proto
+
+require 'google/protobuf'
+
+require 'v2/project_source_pb'
```
Comment (comictvn) [inline ruby/lib/payloads/v2/project_preview_pb.rb:7]: we should remove it manually after codegen. I was faced issue yesterday

Comment (zrg-team) [reply ruby/lib/payloads/v2/project_preview_pb.rb]: Hmm really. Just removed could you check again in your end

**Pull request merged**